### PR TITLE
JAMES-2179 Updating owner right in sharedWith should be filtered out

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java [hunk 1]
```diff
@@ -103,6 +104,9 @@ private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<
         MailboxCounters mailboxCounters = messageManager.getMailboxCounters(mailboxSession);
         MessageManager.MetaData metaData = messageManager.getMetaData(NO_RESET_RECENT, mailboxSession, MessageManager.MetaData.FetchGroup.NO_COUNT);
 
+        System.out.println(messageManager.getMailboxPath());
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java]: syso

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java [hunk 2]
```diff
@@ -206,7 +211,7 @@ public Rights removeEntriesFor(Username username) {
             rights.asMap()
                 .entrySet()
                 .stream()
-                .filter(entry -> entry.getKey().equals(username))
+                .filter(entry -> !entry.getKey().equals(username))
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java]: then how does the previous unit tests work?

Comment (strokyl) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java]: It did not it's just that this should be in the previous commit, therefore I did a fixup for this commit to remove this diff from this commit and an other fixup so this diff get included in the previous commit

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java (hunk 1 above)
Comment (strokyl) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java]: I did a fixup for that too

**Comment by strokyl** [comment]
```
[ERROR] org.apache.james.mailbox.cassandra.mail.CassandraMessageMapperTest.userFlagsShouldBeSupported(org.apache.james.mailbox.cassandra.mail.CassandraMessageMapperTest)
[ERROR]   Run 1: CassandraMessageMapperTest.setUp:66 » IllegalState Can't use this cluster inst...
[ERROR]   Run 2: CassandraMessageMapperTest.tearDown:72 NullPointer
```

**Comment by strokyl** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**